### PR TITLE
[DEV-1787] routes/target: add basic crud endpoints for Target

### PR DESCRIPTION
**Pull request by jevonyeoh**
## Description
Spec: https://featurebyte.atlassian.net/wiki/spaces/DEV/pages/30048257/Target

This PR adds the basic CRUD endpoints for the `Target` API object. Note that this doesn't include any of the actual API object portions yet, and some parts (eg. info, and test structs) are stubbed out for now until we add a bit more functionality (eg. info, forward_join, forward_aggregate, target recipe etc.). Will be working on those in a followup!

## Related Issue
https://featurebyte.atlassian.net/browse/DEV-1787

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: tests/unit/test_generate_payload_fixtures.py [hunk 1]
```diff
@@ -139,6 +140,15 @@ def test_save_payload_fixtures(  # pylint: disable=too-many-arguments
             password="pass",
         ),
     )
+    target = TargetCreate(
+        name="target",
+        graph=feature_sum_30m.graph,
```
Comment (jevonyeoh) [inline tests/unit/test_generate_payload_fixtures.py:145]: placeholders for now, will update w/ a graph from a recipe when that is done

File: featurebyte/models/target.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        Datetime when the Target object was first saved or published
+    updated_at: datetime
+        Datetime when the Target object was last updated
+    """
+
+    # Recipe
+    graph: QueryGraph = Field(allow_mutation=False)
+    node_name: str
+
+    description: Optional[str]
```
Comment (gee-senbong) [inline featurebyte/models/target.py:38]: We should remove description here as there will be another task to add description for all existing assets.

File: featurebyte/models/target.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    # These fields will either be inferred from the recipe, or manually provided by the user only if they're creating
+    # a target without a recipe.
+    window: str
+    blind_spot: str
```
Comment (gee-senbong) [inline featurebyte/models/target.py:43]: Should we add validation logics to `window` & `blind_spot`?

File: .changelog/DEV-1787.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Type of change
+change_type: enhancement
+
+# The name of the component
+component: feature
```
Comment (gee-senbong) [inline .changelog/DEV-1787.yaml:5]: Should we change this to target?

Comment (jevonyeoh) [reply .changelog/DEV-1787.yaml:5]: oh yea, i confused this as "new feature" 😅

File: featurebyte/common/validator.py [hunk 5]
```diff
@@ -145,3 +147,25 @@ def version_validator(cls: Any, value: Any) -> Any:
     if isinstance(value, str):
         return convert_version_string_to_dict(value)
     return value
+
+
+def duration_string_validator(cls: Any, value: Any) -> Any:
+    """
```
Comment (gee-senbong) [inline featurebyte/common/validator.py:153]: Should we reused the logic here: https://github.com/featurebyte/featurebyte/blob/main/featurebyte/common/model_util.py#L15?

File: featurebyte/routes/target/api.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    target_list: TargetList = await controller.list_target(
+        page=page, page_size=page_size, sort_by=sort_by, sort_dir=sort_dir, search=search, name=name
+    )
+    return target_list
+
+
+@router.get("/{target_id}", response_model=TargetModel)
+async def get_target(request: Request, target_id: PydanticObjectId) -> TargetModel:
+    """
+    Retrieve relationship info
```
Comment (gee-senbong) [inline featurebyte/routes/target/api.py:62]: nit: Should we update the docstring here?

File: featurebyte/schema/target.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    data: List[TargetModel]
+
+
+class TargetUpdate(FeatureByteBaseModel):
+    """
+    Target update schema
+    """
+
+    name: StrictStr
+    description: str
```
Comment (gee-senbong) [inline featurebyte/schema/target.py:50]: Should we remove `description` here?

**Review by gee-senbong — APPROVED** [review]
LGTM, would be good to refactor `duration_string_validator` so that same logic can be reused.

File: featurebyte/common/validator.py (hunk 5 above)
Comment (jevonyeoh) [reply featurebyte/common/validator.py:151]: done

File: featurebyte/models/target.py [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        Datetime when the Target object was last updated
+    """
+
+    # Recipe
+    graph: QueryGraph = Field(allow_mutation=False)
+    node_name: str
+
+    # These fields will either be inferred from the recipe, or manually provided by the user only if they're creating
+    # a target without a recipe.
+    window: str
```
Comment (YS-L) [inline featurebyte/models/target.py:41]: Does this refer to the "horizon" field in the spec?

Comment (jevonyeoh) [reply featurebyte/models/target.py:41]: yea you're right! i'll fix this in a follow-up PR

**Pull request merged**